scripts/spack/packages/serac/package.py

```diff
@@ -159,10 +162,18 @@ class Serac(CachedCMakePackage, CudaPackage):
     # CMake packages "build_type=RelWithDebInfo|Debug|Release|MinSizeRel"
 
     # Optional (require our variant in "when")
-    for dep in ["raja", "umpire", "sundials", "strumpack"]:
+    for dep in ["raja", "strumpack"]:
         depends_on("{0} build_type=Debug".format(dep), when="+{0} build_type=Debug".format(dep))
         depends_on("{0}+shared".format(dep), when="+{0}+shared".format(dep))
         depends_on("{0}~shared".format(dep), when="+{0}~shared".format(dep))
+    
+    # Only propagate shared if not CUDA
+    depends_on("umpire build_type=Debug".format(dep), when="+umpire build_type=Debug".format(dep))
+    depends_on("umpire+shared".format(dep), when="+umpire+shared~cuda".format(dep))
+    depends_on("umpire~shared".format(dep), when="+umpire~shared".format(dep))
```

Comment:
Looks a bit ugly bit umpire needed it's own section, due to this conflict: https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/umpire/package.py#L293